build-parent/pom.xml

```diff
@@ -546,27 +546,54 @@
                     </configuration>
                 </plugin>
                 <plugin>
-                    <groupId>io.fabric8</groupId>
+                    <!--
+                        NB: using a temporary fork of the docker-maven-plugin for faster local builds.
+                        This can be removed once https://github.com/fabric8io/docker-maven-plugin/issues/1804
+                        is incorporated into the io.fabric8 plugin directly.
+                      -->
+                    <groupId>org.technologybrewery.fabric8</groupId>
                     <artifactId>docker-maven-plugin</artifactId>
                     <version>${version.fabric8.docker.maven.plugin}</version>
                     <extensions>true</extensions>
+                    <executions>
+                        <execution>
+                            <id>default-push</id>
+                            <phase>deploy</phase>
+                            <goals>
+                                <goal>push</goal>
+                            </goals>
+                            <configuration>
+                                <images>
+                                    <image>
+                                        <build>
+                                            <buildx>
+                                                <builderName>maven</builderName>
+                                                <cacheTo>type=local,dest=~/.docker/cache .</cacheTo>
+                                                <cacheFrom>type=local,src=~/.docker/cache .</cacheFrom>
+                                            </buildx>
+                                        </build>
+                                    </image>
+                                </images>
+                            </configuration>
+                        </execution>
+                    </executions>
                     <configuration>
                         <images>
                             <image>
                                 <name>${docker.baseline.repo.id}/${dockerImageName}:${dockerImageVersion}</name>
                                 <build>
+                                    <args>
+                                        <DOCKER_BASELINE_REPO_ID>${docker.baseline.repo.id}/</DOCKER_BASELINE_REPO_ID>
+                                        <VERSION_AISSEMBLE>${version.aissemble}</VERSION_AISSEMBLE>
+                                        <SPARK_VERSION>${version.spark}</SPARK_VERSION>
+                                    </args>
                                     <buildx>
-                                        <!-- use default cache, passed to docker via config flag.-->
+                                        <builderName>default</builderName>
                                         <dockerStateDir>~/.docker</dockerStateDir>
                                         <platforms>
                                             <platform>${docker.platforms}</platform>
```

Comment:
Q/A: Can we exclude this for this configuration?  Seems like it wouldn't be used, since you can only build the same platform as the host machine.

Reply:
Sadly, if this is not present, the build will trigger as a non-buildx build. To avoid suggesting a more invasive change to fabric8 (and keep it simple), this is the trigger for buildx with the default buildx driver.

Docker does provide lovely, informative error messages if you try to use it out of place.